pvlib/pvsystem.py

```diff
@@ -279,7 +280,7 @@ def physicaliam(self, aoi):
 
         return physicaliam(aoi, **kwargs)
 
-    def calcparams_desoto(self, effective_irradiance, temp_cell, **kwargs):
+    def calcparams_desoto(self, effective_irradiance, temp_cell):
```

Comment:
This is a problem. Method `calcparams_desoto` needs the `**kwargs` because it unpacks these arguments to pass them to function `pvsystem.calcparams_desoto` as positional arguments. 

I don't understand why you are removing `**kwargs` ?

Reply:
Line 302 below assigns directly to kwargs, which (as far as I can tell) clobbers any value for the kwargs dict passed to this method.

Actually, the situation appears similar in the sapm method too, but there the passed kwargs dict is simply never used.

Reply:
To followup further, would you be opposed to forgoing the `_build_kwargs()` call in the `calcparams_desoto` method and using the following (more direct) return statement instead?
```python
return calcparams_desoto(effective_irradiance, temp_cell, **self.module_parameters)
```

Reply:
Removing the `_build_kwargs(...)` step doesn't work given the current setup of `calcparams_desoto`.

Reply:
we need ``_build_kwargs`` for the function call. But I think we can get rid of the ``kwargs`` in the method signature above. Would need to be documented in API changes.